forge_tests/TestFundingCycleWeightCaching.sol

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.6;
+
+import /* {*} from */ "./helpers/TestBaseWorkflow.sol";
+import {MockPriceFeed} from "./mock/MockPriceFeed.sol";
+
+// A funding cycle's weight can be cached to make larger intervals tractible under the gas limit.
+contract TestFundingCycleWeightCaching_Local is TestBaseWorkflow {
+    uint256 private constant _GAS_LIMIT = 30_000_000;
+    uint8 private constant _WEIGHT_DECIMALS = 18; // FIXED
+    uint256 private constant _DURATION = 1;
+    uint256 private constant _DISCOUNT_RATE = 1;
+
+    IJBController3_1 private _controller;
+    IJBFundingCycleStore private _fundingCycleStore;
+    address private _projectOwner;
+
+    JBFundingCycleData private _data;
+    JBFundingCycleMetadata private _metadata;
+
+    function setUp() public override {
+        super.setUp();
+
+        _projectOwner = multisig();
+        _fundingCycleStore = jbFundingCycleStore();
+        _controller = jbController();
+        _data = JBFundingCycleData({
+            duration: _DURATION,
+            weight: 1000 * 10 ** _WEIGHT_DECIMALS,
+            discountRate: _DISCOUNT_RATE,
+            ballot: IJBFundingCycleBallot(address(0))
+        });
+
+        _metadata = JBFundingCycleMetadata({
+            global: JBGlobalFundingCycleMetadata({
+                allowSetTerminals: false,
+                allowSetController: false,
+                pauseTransfers: false
+            }),
+            reservedRate: 0,
+            redemptionRate: 0,
+            baseCurrency: uint32(uint160(JBTokens.ETH)),
+            pausePay: false,
+            allowMinting: false,
+            allowTerminalMigration: false,
+            allowControllerMigration: false,
+            holdFees: false,
+            useTotalOverflowForRedemptions: true,
+            useDataSourceForPay: false,
+            useDataSourceForRedeem: false,
+            dataSource: address(0),
+            metadata: 0
+        });
+    }
+
+    /// Test that caching a cycle's weight yields the same result as computing it.
+    function testWeightCaching(uint256 _cycleDiff) public {
+        // TODO commented out for faster test suite
```

Comment:
How slow was this? I wouldn't remove it from CI (can be in a forge toml CI profile)

Reply:
the test is purposefully testing a case where the loop that decrements a funding cycle's `weight` iteratively for thousands of loops. it's slow on purpose.